src/python/tests/gff3/test_extract_annotation.py

```diff
@@ -378,3 +402,53 @@ def test_store_gene(
     assert len(annot.features["gene"]) == expected_num_genes
     assert len(annot.features["transcript"]) == expected_num_tr
     assert len(annot.features["translation"]) == expected_num_cds
+
+
+@pytest.mark.parametrize(
+    "gene, transcript, translation, expected_json",
+    [
+        pytest.param(
+            SeqFeature(type="gene", id="gene_A"),
+            SeqFeature(type="mRNA", id="tran_A"),
+            SeqFeature(type="CDS", id="cds_A"),
+            "dump_noinfo.json",
+            id="No annotation",
+        ),
+        pytest.param(
+            SeqFeature(
+                type="gene", id="gene_A", qualifiers={"description": ["Gene description"], "Name": ["GeneA"]}
+            ),
+            SeqFeature(type="mRNA", id="tran_A"),
+            SeqFeature(type="CDS", id="cds_A"),
+            "dump_syn.json",
+            id="Some annotation",
+        ),
+        pytest.param(
+            SeqFeature(type="gene", id="gene_A", qualifiers={"description": ["Gene NameA"]}),
+            SeqFeature(type="mRNA", id="tran_A", qualifiers={"description": ["Transcript NameA"]}),
+            SeqFeature(type="CDS", id="cds_A", qualifiers={"description": ["Protein NameA"]}),
+            "dump_description.json",
+            id="Some descriptions",
+        ),
+    ],
+)
+def test_to_json(
+    assert_files: Callable,
+    tmp_dir: Path,
```

Comment:
```suggestion
    tmp_path: Path,
```
I definitely need to get rid of `tmp_dir` in `ensembl-py` asap.